components/org.wso2.carbon.identity.organization.management.handler/src/test/java/org/wso2/carbon/identity/organization/management/handler/tests/GovernanceConfigUpdateHandlerTest.java

```diff
@@ -0,0 +1,87 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.organization.management.handler.tests;
+
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.governance.IdentityGovernanceException;
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.organization.management.ext.Constants;
+import org.wso2.carbon.identity.organization.management.handler.GovernanceConfigUpdateHandler;
+import org.wso2.carbon.identity.organization.management.handler.internal.OrganizationManagementHandlerDataHolder;
+import org.wso2.carbon.identity.organization.management.service.OrganizationManager;
+import org.wso2.carbon.identity.organization.management.service.exception.OrganizationManagementException;
+import org.wso2.carbon.identity.organization.management.service.model.Organization;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class GovernanceConfigUpdateHandlerTest {
+
+    @Mock
+    private IdentityGovernanceService identityGovernanceService;
+
+    @Mock
+    private OrganizationManager organizationManager;
+
+    @InjectMocks
+    private GovernanceConfigUpdateHandler configUpdateHandler;
+
+    @BeforeMethod
+    public void setUp() {
+
+        MockitoAnnotations.openMocks(this);
+        OrganizationManagementHandlerDataHolder.getInstance().setIdentityGovernanceService(identityGovernanceService);
+        OrganizationManagementHandlerDataHolder.getInstance().setOrganizationManager(organizationManager);
+    }
+
+    @Test
+    public void testHandleEventWithPostAddOrganizationEvent()
+            throws IdentityEventException, OrganizationManagementException, IdentityGovernanceException {
+
+        // Create a sample organization.
+        Organization organization = new Organization();
+        organization.setId("orgId");
+
+        // Mock the necessary methods.
+        when(organizationManager.resolveTenantDomain(organization.getId())).thenReturn("sampleTenantDomain");
+        when(organizationManager.getOrganizationDepthInHierarchy(organization.getId())).thenReturn(2);
```

Comment:
```suggestion
        when(organizationManager.getOrganizationDepthInHierarchy(organization.getId())).thenReturn(1);
```
Except IDaaS deployments, suborganizations will appear at depth 1. Hence better to check for depth 1.

Reply:
Resolved in [5e22e0e](https://github.com/wso2-extensions/identity-organization-management/pull/205/commits/5e22e0e447e3b37f0f14f6a5aa18e9510faf4a4b)